### ISSUE TITLE
Tool not calcualting statistics.

### ISSUE THREAD

**Issue by Mattah12** [juliepierson/DissolveWithStats#10]
The tool is just returning the values for the first record in each dissolve field value, rather than calcualting statistics.

It still carries out the physical dissolve.

For example when carrying out a dissolve on the below with 'Sum' calculated as the statistic:

![image](https://user-images.githubusercontent.com/4283530/214616421-8f59f0cb-4abe-4094-9023-31adc5b6646f.png)

It returns just the first record:
![image](https://user-images.githubusercontent.com/4283530/214616682-d948a194-b595-400f-88a7-c388d13d97fd.png)

**Comment by juliepierson** [comment]
Hi @Mattah12,
Thanks for your feedback !

First of all this tool is no longer maintained, since there is now a core qgis tool which does the job : [aggregate](https://docs.qgis.org/3.22/en/docs/user_manual/processing_algs/qgis/vectorgeometry.html#aggregate)

Still it should work :-) Could you send me your data and a screenshot of the tool with all your parameters ?
Thanks !

**Comment by Mattah12** [comment]
Thanks for the prompt reply. I had no idea about Aggregate, does the job perfectly.

My data is saved here - https://drive.google.com/file/d/1BSFIazR3-TznuWvZn9GIw_UVQzAUxQqr/view?usp=share_link

And below is how I'm using the tool:

![image](https://user-images.githubusercontent.com/4283530/214811450-fd715607-a5c1-48a4-9dcc-00a452677911.png)

You can't see the first 6 fields, but I'm not keeping those anyway.

I just ran it again to be sure, and it's still just spitting out the first feature of each of the 'name' fields.

**Comment by juliepierson** [comment]
Ok, thanks, I'll have a look into this as soon as I can !

**Comment by juliepierson** [comment]
I just tried it, I had a warning message during dissolve with stats "cannot delete identifier column" (or something like that, message was in french). I tried again after exporting your layer in a shapefile without fid column, I had to repair geometries first and it worked all right.

When I created this plugin, geopackage wasn't much around (it was some time ago !), and the fid automatic field does not seem to be well handled.

Of course it's better to use the core aggregate tool :-)
